src/routes/committee/announcements/+page.svelte

```diff
@@ -0,0 +1,162 @@
+<script lang="ts">
+	import {
+		Grid,
+		Row,
+		Column,
+		StructuredList,
+		StructuredListCell,
+		StructuredListRow,
+		Button,
+		Modal,
+		Form,
+		TextInput,
+		TextArea
+	} from 'carbon-components-svelte'
+	import type { announcement, newAnnouncement } from '$lib/types/domain/announcement'
+	import { Add, TrashCan } from 'carbon-icons-svelte'
+	import { addDoc, collection, deleteDoc, doc } from 'firebase/firestore'
+	import { page } from '$app/stores'
+	import { db } from '$lib/firebase/client/firebase'
+	import { Timestamp } from 'firebase/firestore'
+	import { newNewsPost } from '$lib/util/webhook'
+
+	export let data: { announcements: announcement[] }
+
+	let openNew = false
+	let openDel = false
+	let title: string
+	let message: string
+	let selectedAnnouncement: number = 0
+
+	console.log(data.announcements)
+
+	async function AddAnnouncement() {
+		const user = doc(db, 'users', $page.data.user!.databaseId)
+
+		// TODO: use current edition
+		const edition = doc(db, 'editions', 'ZI3Eab1mXjHvCUS47o40')
+		const publishDate = Timestamp.fromDate(new Date())
+
+		const newAnnouncement: newAnnouncement = {
+			edition,
+			publishDate,
+			title,
+			content: message,
+			authorUser: user
+		}
+
+		let newAnnouncementRef = await addDoc(collection(db, 'announcements'), newAnnouncement)
+
+		newNewsPost(newAnnouncement)
+		let newAnnouncementElement: announcement = {
+			id: newAnnouncementRef.id,
+			...newAnnouncement
+		} as announcement
+
+		data.announcements = [newAnnouncementElement, ...data.announcements]
+	}
+
+	async function RemoveAnnouncement() {
+		const announcementRef = doc(db, 'announcements', data.announcements[selectedAnnouncement].id)
+		await deleteDoc(announcementRef)
+
+		data.announcements.splice(selectedAnnouncement, 1)
+		data.announcements = data.announcements
```

Comment:
Ik zou deze ook met invalidate doen. Is wat nicer omdat je dan nooit problemen met index gaat hebben, en je weet meteen zeker dat je pagina weer up to date is met de db